### PR TITLE
[BREAKING][data, trainer] refactor: make NeoProto the default V0 DataProto

### PR DESCRIPTION
**Pull request by alanSquirrelyz**
### What does this PR do?

  This PR makes the NeoProto-backed `DataProto` the public data representation used by the synchronous V0 `RayPPOTrainer`.

  The existing `DataProto` API remains available through `from verl import DataProto`, while the previous TensorDict-backed implementation remains explicitly accessible as `verl.LegacyDataProto` and
  `verl.protocol.DataProto`.

  Storage remains pluggable. The V0 Ray trainer uses the Ray object store by default, while callers can provide another `StorageEngine` through the existing construction APIs.

  This PR also removes the transitional runtime data-plane selector, redundant adapter layers, and correctness-only dump scripts that are no longer needed once the trainer has a single `DataProto`
  implementation.

  Related work:

  - #7350 introduced the initial NeoProto ref/index data path and was merged into `neo`. This PR builds directly on that merged commit and removes the transitional selectable data-plane abstraction.
  - #7000 focuses on removal of the previous transfer-queue path and CI preparation; it does not provide the public `DataProto` compatibility layer or this storage-backed V0 integration.
  - RFC #6988 provides the broader design context.

  This is not a duplicate of an existing open PR: it consolidates the already-merged NeoProto work into the V0 trainer's public `DataProto` path while retaining explicit legacy compatibility.

  ### Checklist Before Starting

  - [x] Search for similar PRs:
    - https://github.com/verl-project/verl/pulls?q=is%3Apr+is%3Aopen+NeoProto
    - https://github.com/verl-project/verl/pulls?q=is%3Apr+is%3Aopen+%22data+plane%22+%22storage+backend%22

  Test Result:

  - Completed two full training steps successfully on a single 8-GPU node.
  - Rollout generation, actor update, checkpoint creation, and final validation completed.

  ### API and Usage Example

  Existing public imports continue to work:

```
from verl import DataProto

batch = DataProto.from_dict(
    tensors=tensors,
    non_tensors=non_tensors,
    meta_info=meta_info,
)
```

  verl.DataProto is now backed by NeoProto refs and index views. Existing batch operations such as select, pop, union, chunk, concat, repeat, and __getitem__ remain supported.

  A storage engine may be supplied explicitly:

```
from verl import DataProto
from verl.experimental.neoproto.storage import InMemoryStorageEngine

batch = DataProto.from_dict(
    tensors=tensors,
    non_tensors=non_tensors,
    storage=InMemoryStorageEngine(),
)
```

  The previous implementation remains available for explicit compatibility use:

```
from verl import LegacyDataProto

legacy_batch = LegacyDataProto.from_dict(
    tensors=tensors,
    non_tensors=non_tensors,
)
```
  Direct imports from verl.protocol also continue to resolve to the legacy implementation:

```
from verl.protocol import DataProto

The following transitional trainer settings are removed and must no longer be passed:

trainer.data_plane
trainer.use_neoproto
trainer.neoproto_strict_mode
```

  ### Design & Code Changes
 - Export the NeoProto-backed implementation as `verl.DataProto`, while retaining the previous implementation as `verl.LegacyDataProto`.
  - Use NeoProto directly in the synchronous V0 `RayPPOTrainer` and remove the transitional runtime data-plane selector.
  - Keep storage pluggable, with the Ray object store as the trainer default.
  - Remove obsolete compatibility scaffolding, correctness-only scripts, and configuration options.

  Training datasets, rollout inputs, reward computation, PPO algorithms, and checkpoint formats are not changed by this refactor.

  ### Checklist Before Submitting

  - [x] Apply pre-commit checks.
  - [x] Update the NeoProto integration documentation.
  - [x] Update unit and end-to-end test coverage for the consolidated path.

### CODE REVIEW

**Review by zw0610 — APPROVED** [review]
/LGTM
